app/models/comparison/metric.rb

```diff
@@ -0,0 +1,49 @@
+# == Schema Information
+#
+# Table name: comparison_metrics
+#
+#  alert_type_id             :bigint(8)        not null
+#  asof_date                 :date
+#  created_at                :datetime         not null
+#  custom_period_id          :bigint(8)
+#  custom_previous_period_id :bigint(8)
+#  enough_data               :boolean          not null
+#  id                        :bigint(8)        not null, primary key
+#  metric_type_id            :bigint(8)        not null
+#  recent_data               :boolean          not null
+#  reporting_period          :integer
+#  school_id                 :bigint(8)        not null
+#  updated_at                :datetime         not null
+#  value                     :string
+#  whole_period              :boolean          not null
+#
+# Indexes
+#
+#  index_comparison_metrics_on_alert_type_id              (alert_type_id)
+#  index_comparison_metrics_on_custom_period_id           (custom_period_id)
+#  index_comparison_metrics_on_custom_previous_period_id  (custom_previous_period_id)
+#  index_comparison_metrics_on_metric_type_id             (metric_type_id)
+#  index_comparison_metrics_on_school_id                  (school_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (custom_period_id => comparison_periods.id) ON DELETE => cascade
+#  fk_rails_...  (custom_previous_period_id => comparison_periods.id) ON DELETE => cascade
+#  fk_rails_...  (metric_type_id => comparison_metrics.id) ON DELETE => cascade
+#  fk_rails_...  (school_id => schools.id) ON DELETE => cascade
+#
+class Comparison::Metric < ApplicationRecord
+  self.table_name = 'comparison_metrics'
+
+  # Add reporting periods as required?
+  # [:last_12_months, :financial_year, :academic_year]
+  enum reporting_period: { custom: 0 }
```

Comment:
We could add `last_12_months` now, as it'll be needed to start populating some actual values. The `custom` periods will be an additional new feature. The others will get added as we do them. At the moment its looking like `school_year` is the one most used by schools. 

If the period is `custom` then we should have a `custom_period` defined. So could add a validation for that.